plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -258,6 +263,14 @@ private static ConnectorPageSource createOrcPageSource(
                 return new EmptyPageSource();
             }
             OrcReader reader = optionalOrcReader.get();
+            if (!originalFile && acidInfo.isPresent() && !acidInfo.get().isOrcAcidVersionValidated()) {
+                // Trino cannot read ORC ACID tables with version < 2 (written by Hive older than 3.0)
+                // See https://github.com/trinodb/trino/issues/2790#issuecomment-591901728 for more context
+
+                // If we did not manage to validate if ORC ACID version used by table is supported one base don _orc_acid_version metadata file
+                // we check the data file footer.
```

Comment:
You can move this explanation into validateOrcAcidVersion, somewhere before throw